src/tools/miri/src/concurrency/weak_memory.rs

```diff
@@ -82,7 +82,6 @@ use std::{
     collections::VecDeque,
 };
 
-use rustc_const_eval::interpret::{alloc_range, AllocRange, InterpResult, MPlaceTy, Scalar};
```

Comment:
These types are not in the prelude. So if the lint fires here then redundant_prelude_imports seems like a confusing name?

Reply:
> These types are not in the prelude. So if the lint fires here then redundant_prelude_imports seems like a confusing name?

Yes, according to petrochenkov's review comments and help, the redundant prelude lint was not added, and instead edit lint unused_imports which is checked by checked_for_redundant_imports in rustc_resolve. I think I should change the description and title of the PR.